### PR TITLE
Fix sphinx docstring warnings and add spec docs

### PR DESCRIPTION
**Pull request by jbeezley**
The api docs could probably be better organized with a bit of an introduction for each component, but at least includes them.

### CODE REVIEW

**Comment by danlamanna** [comment]
LGTM.

(I didn't get any docstring warnings before, running ```Sphinx==1.3.1```)

**Comment by jbeezley** [comment]
You have to blow out your _build directory because the docstrings don't ever seem to get rebuilt.  I always just do `make clean html`.

File: docs/types-and-formats.rst [hunk 1]
```diff
@@ -93,7 +93,6 @@ A list of rows with ordered, named column attributes. Formats:
 :``"csv"``: A string containing the contents of a comma-separated CSV file.
     Column headers will be reasonably detected if present, otherwise
     columns will be named ``"Column 1"``, ``Column 2"``, etc.
-    See `has_header`_ for details on header detection.
```
Comment (jbeezley) [inline docs/types-and-formats.rst]: @jeffbaumes I removed this line because the link is invalid, and I couldn't make any sense of it.

**Comment by danlamanna** [comment]
Reproduced, this fixes it. #justShipIt

File: docs/types-and-formats.rst (hunk 1 above)
Comment (jeffbaumes) [inline docs/types-and-formats.rst]: Yeah that's old stuff. I believe column headers are now assumed - so lines 94-95 should be removed as well.

File: docs/types-and-formats.rst [hunk 2]
```diff
@@ -91,9 +91,7 @@ A list of rows with ordered, named column attributes. Formats:
     ``"objectlist"`` format. This is the format of MongoDB collections.
 
 :``"csv"``: A string containing the contents of a comma-separated CSV file.
-    Column headers will be reasonably detected if present, otherwise
-    columns will be named ``"Column 1"``, ``Column 2"``, etc.
-    See `has_header`_ for details on header detection.
+    The first line of the file is assumed to contain column headers.
```
Comment (jbeezley) [inline docs/types-and-formats.rst]: @jeffbaumes How is this?

Comment (jeffbaumes) [inline docs/types-and-formats.rst]: Excellent, thanks.

**Pull request merged**